be/src/agent/agent_server.cpp

```diff
@@ -409,13 +409,12 @@ void AgentServer::make_snapshot(TAgentResult& return_value,
     TStatus status;
     vector<string> error_msgs;
     TStatusCode::type status_code = TStatusCode::OK;
-    int32_t return_snapshot_version = PREFERRED_SNAPSHOT_VERSION;
-    // if the request's snapshot version is less than current be's snapshot version
-    // it means the request be is under old version. just set the request version to 1
-    // current be will generate snapshot files like tabletid_schemahash_startversion_endversion
-    // format. Every be is able to parse this format snapshot files.
-    if (snapshot_request.preferred_snapshot_version <  PREFERRED_SNAPSHOT_VERSION) {
-        return_snapshot_version = 1;
+    int32_t return_snapshot_version = BETA_ROWSET_VERSION;
+    // If the request's preferred snapshot version is ALPHA,
+    // Then we will convert the BETA to ALPHA
+    // So, set the return_snapshot_version to ALPHA
+    if (snapshot_request.preferred_snapshot_version == ALPHA_ROWSET_VERSION) {
+        return_snapshot_version = ALPHA_ROWSET_VERSION;
     }
```

Comment:
If snapshot_request.preferred_snapshot_version less than ALPHA_ROWSET_VERSION, better to return error to client to make it know we don't support its format any more.

Reply:
Error is returned by set make_snapshot_status to be not OLAP_SUCCESS. It has been done in SnapshotManager